libobs-d3d11/d3d11-shader.cpp

```diff
@@ -224,29 +224,61 @@ void gs_shader::Compile(const char *shaderString, const char *file,
 	ComPtr<ID3D10Blob> errorsBlob;
 	HRESULT hr;
 
+	bool is_cached = false;
 	char hashstr[20];
 
 	if (!shaderString)
 		throw "No shader string specified";
 
-	uint64_t hash = fnv1a_hash(shaderString);
+	uint64_t hash = fnv1a_hash(shaderString, strlen(shaderString));
```

Comment:
We also use `strlen(shaderString)` in the non-cached path, so this should probably be stored in a variable.